src/execute.ts

```diff
@@ -24,122 +24,137 @@ export default function execute<T>(request: httpclient.Request): Promise<httpcli
 
 		// Creating a new XMLHttpRequest and giving it two properties from the request this method received
 		// This will be the request we will send to the server
-		const xhr = new XMLHttpRequest()
-		xhr.withCredentials = request.withCredentials
-		xhr.timeout = request.timeout
+		try {
+			const xhr = new XMLHttpRequest()
+			request.xhr = xhr
```

Comment:
Check if request is aborted here and reject() if it is the case. That way, you can remove the try/catch you added

Reply:
ok